bundles/org.openhab.binding.juicenet/src/main/java/org/openhab/binding/juicenet/internal/handler/JuiceNetDeviceHandler.java

```diff
@@ -0,0 +1,349 @@
+                    Long targetTime = datetime.toEpochSecond() + datetime.get(ChronoField.OFFSET_SECONDS);
+                    logger.debug("DateTime: {} - {}", datetime.toString(), targetTime);
+
+                    getApi().setOverride(Objects.requireNonNull(token), energyAtPlugin, targetTime, energyToAdd);
+
+                    break;
+                }
+                case CHANNEL_CHARGING_STATE: {
+                    String state = ((StringType) command).toString();
+                    Long overrideTime = deviceStatus.unitTime;
+                    int energyAtPlugin = 0;
+                    int energyToAdd = deviceCar.batterySize;
+
+                    switch (state) {
+                        case "stop":
+                            if (targetTimeTou == 0) {
+                                targetTimeTou = deviceStatus.targetTime;
+                            }
+                            overrideTime = deviceStatus.unitTime + 31556926;
+                            break;
+                        case "start":
+                            if (targetTimeTou == 0) {
+                                targetTimeTou = deviceStatus.targetTime;
+                            }
+                            overrideTime = deviceStatus.unitTime;
+                            break;
+                        case "smart":
+                            overrideTime = deviceStatus.defaultTargetTime;
+                            break;
+                    }
+
+                    getApi().setOverride(Objects.requireNonNull(token), energyAtPlugin, overrideTime, energyToAdd);
+
+                    break;
+                }
+            }
+        } catch (JuiceNetApiException | InterruptedException e) {
+            handleApiException(e);
+            return;
+        }
+    }
+
+    public void tryQueryDeviceStatusAndInfo() throws JuiceNetApiException, InterruptedException {
+        String apiToken = Objects.requireNonNull(this.token);
+
+        deviceStatus = getApi().queryDeviceStatus(apiToken);
+
+        if (deviceStatus.timestamp > lastInfoTimestamp) {
+            lastInfoTimestamp = deviceStatus.timestamp;
+
+            deviceInfo = getApi().queryInfo(apiToken);
+            deviceTouSchedule = getApi().queryTOUSchedule(apiToken);
+            refreshInfoChannels();
+        }
+
+        int carId = deviceStatus.carId;
+        for (JuiceNetApiCar car : deviceInfo.cars) {
+            if (car.carId == carId) {
+                this.deviceCar = car;
+                break;
+            }
+        }
+
+        refreshStatusChannels();
+    }
+
+    public void queryDeviceStatusAndInfo() {
+        logger.trace("queryStatusAndInfo");
+        ThingStatus status = getThing().getStatus();
+
+        if (status == ThingStatus.OFFLINE || status == ThingStatus.UNKNOWN) {
+            goOnline();
+            return;
+        }
+
+        if (this.getThing().getStatus() != ThingStatus.ONLINE) {
+            return;
+        }
```

Comment:
For me this is hard to read, not sure if tricky or wrong. If offline/unknown, go online and return. If online, skip both conditionals. If other statuses, return. Is this some trickery for handling uninitialized, initializing etc. to not try anything?

Reply:
yeah - this was leftover code.  It is redundant.

Reply:
Still wondering if `status != ThingStatus.ONLINE` would be a safer condition here?